packages/fx-core/src/plugins/resource/apim/managers/questionManager.ts

```diff
@@ -1,17 +1,13 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
-import { Func, PluginContext, QTreeNode } from "@microsoft/teamsfx-api";
-import { BuildError, NotImplemented } from "../error";
+import { EnvInfo, Func, PluginContext, QTreeNode, v3 } from "@microsoft/teamsfx-api";
+import { BuiltInFeaturePluginNames } from "../../../solution/fx-solution/v3/constants";
 import { IApimPluginConfig } from "../config";
-import * as VSCode from "../questions/vscodeQuestion";
+import { BuildError, NotImplemented } from "../error";
 import * as CLI from "../questions/cliQuestion";
+import * as VSCode from "../questions/vscodeQuestion";
 
-export interface IQuestionManager {
```

Comment:
Can we keep the interface here and remove the logic to check the type of question manager?
```
 switch (stage) {
    case Stage.deploy:
      return questionManager instanceof VscQuestionManager
        ? await (questionManager as VscQuestionManager).deploy(ctx.root, ctx.envInfo, apimConfig)
        : await (questionManager as CliQuestionManager).deploy();
    default:
      return undefined;
```